Pinta.Effects/Effects/AlignObjectEffect.cs

```diff
@@ -0,0 +1,169 @@
+			Data.Position = dialog.SelectedPosition;
+		};
+
+		dialog.OnResponse += (_, args) => {
+			OnConfigDialogResponse (args.ResponseId == (int) Gtk.ResponseType.Ok);
+			dialog.Destroy ();
+		};
+
+		dialog.Present ();
+	}
+
+	public override void Render (ImageSurface src, ImageSurface dest, ReadOnlySpan<RectangleI> rois)
+	{
+		// If no selection, it's the whole image
+		RectangleI selection = rois[0];
+		AlignPosition align = Data.Position;
+
+		RectangleI objectBounds = Utility.GetObjectBounds (src, selection);
+
+		// Calculate the new position for the object
+		PointI newPosition = CalculateNewPosition (objectBounds, align, selection);
+
+		// Draw the object in the new position
+		MoveObject (src, dest, objectBounds, newPosition, selection);
+	}
+
+	private PointI CalculateNewPosition (RectangleI objectBounds, AlignPosition align, RectangleI selectionBounds)
+	{
+		int x = 0;
+		int y = 0;
+
+		// Align with the selection bounds
+		switch (align) {
+			case AlignPosition.TopLeft:
+				x = selectionBounds.X;
+				y = selectionBounds.Y;
+				break;
+			case AlignPosition.TopCenter:
+				x = selectionBounds.X + selectionBounds.Width / 2 - objectBounds.Width / 2;
+				y = selectionBounds.Y;
+				break;
+			case AlignPosition.TopRight:
+				x = selectionBounds.Right - objectBounds.Width;
+				y = selectionBounds.Y;
+				break;
+			case AlignPosition.CenterLeft:
+				x = selectionBounds.X;
+				y = selectionBounds.Y + selectionBounds.Height / 2 - objectBounds.Height / 2;
+				break;
+			case AlignPosition.Center:
+				x = selectionBounds.X + selectionBounds.Width / 2 - objectBounds.Width / 2;
+				y = selectionBounds.Y + selectionBounds.Height / 2 - objectBounds.Height / 2;
+				break;
+			case AlignPosition.CenterRight:
+				x = selectionBounds.Right - objectBounds.Width;
+				y = selectionBounds.Y + selectionBounds.Height / 2 - objectBounds.Height / 2;
+				break;
+			case AlignPosition.BottomLeft:
+				x = selectionBounds.X;
+				y = selectionBounds.Bottom - objectBounds.Height;
+				break;
+			case AlignPosition.BottomCenter:
+				x = selectionBounds.X + selectionBounds.Width / 2 - objectBounds.Width / 2;
+				y = selectionBounds.Bottom - objectBounds.Height;
+				break;
+			case AlignPosition.BottomRight:
+				x = selectionBounds.Right - objectBounds.Width;
+				y = selectionBounds.Bottom - objectBounds.Height;
+				break;
+		}
+
+		return new PointI (x, y);
+	}
+
+	private void MoveObject (ImageSurface src, ImageSurface dest, RectangleI objectBounds, PointI newPosition, RectangleI selectionBounds)
+	{
+		var src_data = src.GetReadOnlyPixelData ();
+		var dst_data = dest.GetPixelData ();
+		int width = src.Width;
+
+		// Clear the selection area
+		for (int y = 0; y < selectionBounds.Height; y++) {
+			var dst_row = dst_data.Slice ((selectionBounds.Y + y) * width + selectionBounds.X, selectionBounds.Width);
+
+			for (int i = 0; i < dst_row.Length; i++) {
+				dst_row[i] = src.GetColorBgra (PointI.Zero);
```

Comment:
This fills the selection with the color from pixel (0,0) of the image, but I think the intent is to just fill it with transparent pixels? There's a chance that source pixel might not be transparent if it's outside of the selection

Also, I think the loop could be replaced by `dst_row.Fill()`?

Reply:
The intent is actually to pick the 0,0 color, since that's supposed to be the "background" color, which is also used for object detection.
This means object detection also works with non-transparent backgrounds, although that's not a particularly useful feature.
dst_row.Fill() seems to work, i replaced the call with that.

Reply:
makes sense, done!